libvirt/resource_libvirt_domain.go

```diff
@@ -698,6 +704,8 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	defer destroyDomainByUserRequest(d, domain)
```

Comment:
@monstermunchkin  why we don't need `defer` here? imho is more elegant to use it. 
I mean this destroying function is a `cleanup` function

Reply:
What @monstermunchkin means it that you want to call destroyDomainByUserRequest _there_ and not later. Any programmer reading the code would think "why is he deferring something he wants to do now?".
While in the other example `defer domain.Free()` you are indeed intending "Call `domain.Free()` once all the rest of the code is executed, and _not_ before, because I will be using `domain`".

Reply:
@dmacvicar @monstermunchkin  thx for clarification. 
I put `defer` there, because we are shutting down a libvirt domain, and imho it should be added later like `defer domain.Free`, for one reason only: 
- imagine the domain is shut off (down). and people are adding after that function other operations on domain. 
The question is , are all domain operation possible on a shutdown domain without errors? 
So that's why i put this in a `defer` because to me seemed a function that need to go @ end